disperser/apiserver/server_v2.go

```diff
@@ -319,25 +336,41 @@ func (s *DispersalServerV2) RefreshOnchainState(ctx context.Context) error {
 	return nil
 }
 
-func (s *DispersalServerV2) GetPaymentState(ctx context.Context, req *pb.GetPaymentStateRequest) (*pb.GetPaymentStateReply, error) {
+func (s *DispersalServerV2) GetPaymentState(
+	ctx context.Context,
+	req *pb.GetPaymentStateRequest,
+) (*pb.GetPaymentStateReply, error) {
+	reply, st := s.getPaymentState(ctx, req)
+	api.LogResponseStatus(s.logger, st)
+	if st != nil {
+		// nolint:wrapcheck
+		return reply, st.Err()
+	}
+	return reply, nil
+}
+
+func (s *DispersalServerV2) getPaymentState(
+	ctx context.Context,
+	req *pb.GetPaymentStateRequest,
+) (*pb.GetPaymentStateReply, *status.Status) {
 	if s.meterer == nil {
-		return nil, errors.New("payment meterer is not enabled")
+		return nil, status.New(codes.Internal, "meterer is not configured")
 	}
 	start := time.Now()
 	defer func() {
 		s.metrics.reportGetPaymentStateLatency(time.Since(start))
 	}()
 
 	if !gethcommon.IsHexAddress(req.GetAccountId()) {
-		return nil, api.NewErrorInvalidArg("invalid account ID")
+		return nil, status.New(codes.InvalidArgument, "invalid account ID")
 	}
 
 	accountID := gethcommon.HexToAddress(req.GetAccountId())
 
 	// validate the signature
 	if err := s.blobRequestAuthenticator.AuthenticatePaymentStateRequest(accountID, req); err != nil {
 		s.logger.Debug("failed to validate signature", "err", err, "accountID", accountID)
-		return nil, api.NewErrorInvalidArg(fmt.Sprintf("authentication failed: %s", err.Error()))
+		return nil, status.Newf(codes.Unauthenticated, "failed to validate signature: %s", err.Error())
```

Comment:
You changed from `InvalidArgument` to `Unauthenticated`. Was this intentional?

Reply:
Unauthenticated makes sense to me, just wanted to make sure it wasn't inadvertent